sql/scan.go

```diff
@@ -74,11 +73,9 @@ type span struct {
 // prettyKey pretty-prints the specified key, skipping over the first skip
 // fields.
 func prettyKey(key roachpb.Key, skip int) string {
-	if !bytes.HasPrefix(key, keys.TableDataPrefix) {
-		return fmt.Sprintf("index key missing table data prefix: %q vs %q",
-			key, keys.TableDataPrefix)
+	if encoding.PeekType(key) != encoding.Int {
+		return fmt.Sprintf("index key missing table data prefix: %q", key)
```

Comment:
Adjusted the message to not mention "table data", but I'm not sure what else you want. I want to see the key from which I'll be able to see that the first byte is not a valid int tag.